tgui/packages/tgui/interfaces/ExodroneConsole.tsx

```diff
@@ -0,0 +1,793 @@
+
+  return (
+    <Window width={650} height={500}>
+      {!!signal_lost && <SignalLostModal />}
+      {!!choosingTools && <ToolSelectionModal />}
+      <Window.Content>
+        <ExodroneConsoleContent />
+      </Window.Content>
+    </Window>
+  );
+};
+
+const SignalLostModal = (props, context) => {
+  const { act } = useBackend(context);
+  return (
+    <Modal backgroundColor="red" textAlign="center" width={30} height={22} p={0} style={{ "border-radius": "5%" }}>
+      <img src={nt_logo} width={64} height={64} />
+      <Box backgroundColor="black" textColor="red" fontSize={2} style={{ "border-radius": "-10%" }}>CONNECTION LOST</Box>
+      <Box p={2} italic>
+        Connection to exploration drone interrupted.
+        Please contact nearest Nanotrasen Exploration Division
+        representative for further instructions.
+      </Box>
+      <Icon name="exclamation-triangle" textColor="black" size={5} />
+      <Box><Button color="danger" style={{ "border": "1px solid black" }} onClick={() => act("confirm_signal_lost")}>Confirm</Button></Box>
+    </Modal>);
+};
+
+const DroneSelectionSection = (props, context) => {
+  const { act, data } = useBackend<ExodroneConsoleData>(context);
+  const {
+    all_drones,
+  } = data;
+
+  return (
+    <Section scrollable fill title="Exploration Drone Listing">
+      <Stack vertical>
+        {all_drones.map(drone => (
+          <>
+            <Stack.Item grow key={drone.ref}>
+              <Stack fill>
+                <Stack.Item basis={10} fontFamily="monospace" fontSize="18px">
+                  {drone.name}
+                </Stack.Item>
+                <Stack.Divider />
+                <Stack.Item fontFamily="monospace" mt={0.8}>
+                  {drone.description}
+                </Stack.Item>
+                <Stack.Item grow />
+                <Stack.Divider mr={1} />
+                <Stack.Item ml={0}>
+                  {drone.controlled && (
+                    "Controlled by another console."
+                  ) || (
+                    <Button
+                      content="Assume Control"
+                      icon="plug"
+                      onClick={() => act("select_drone", { "drone_ref": drone.ref })} />
+                  )}
+                </Stack.Item>
+              </Stack>
+            </Stack.Item>
+            <Stack.Divider />
+          </>
+        ))}
+      </Stack>
+    </Section>);
+};
+
+
+const ToolSelectionModal = (props, context) => {
+  const { act, data } = useBackend<ExodroneConsoleData>(context);
+  const {
+    all_tools = {},
+  } = data;
+
+  const [
+    choosingTools,
+    setChoosingTools,
+  ] = useLocalState(context, 'choosingTools', false);
+
+  return (
+    <Modal>
+      <Stack fill vertical pr={2}>
+        <Stack.Item>
+          Select Tool:
+        </Stack.Item>
+        <Stack.Item>
+          <Stack textAlign="center">
+            {!!Object.keys(all_tools) && Object.keys(all_tools).map(tool_name => (
```

Comment:
if you need help with tabbing out maps, this is a perfect example of one (and the indentation is right too)